x/upgrade/plan/info.go

```diff
@@ -77,10 +80,12 @@ func (m BinaryDownloadURLMap) ValidateBasic() error {
 	}
 
 	osArchRx := regexp.MustCompile(`[a-zA-Z0-9]+/[a-zA-Z0-9]+`)
-	for key, val := range m {
+	for _, key := range maps.Keys(m) {
 		if key != "any" && !osArchRx.MatchString(key) {
 			return fmt.Errorf("invalid os/arch format in key \"%s\"", key)
 		}
+
+		val := m[key]
```

Comment:
This code isn't sorting the keys, so AFAICT doesn't produce behavior any different than ranging over the map directly?